pulsar/message_chunking_test.go

```diff
@@ -548,30 +556,57 @@ func createTestMessagePayload(size int) []byte {
 }
 
 //nolint:all
-func sendSingleChunk(p Producer, uuid string, chunkID int, totalChunks int) {
+func sendSingleChunk(p Producer, uuid string, chunkID int, totalChunks int, wholePayload string, callbackOnce *sync.Once, cr *chunkRecorder) {
```

Comment:
Should we add the parameters`wholePayload ` `callbackOnce ` and `chunkRecorder `? I think it's more appropriate to make them inside the`sendSingleChunk `.

Reply:
Hmm, in the current implemention, all the chunking sendRequest should share the whole message payload, callbackOnce, chunkRecorder, see `internalSend()` and `addRequestToBatch()`, I add these params just want to make the send procedure work correctly, I think it better to keep them.